src/metabase/query_processor/util/nest_query.clj

```diff
@@ -124,6 +124,20 @@
                 ::add/source-table  ::add/source
                 ::add/source-alias  source-alias))]))
 
+(defn- coerced-field?
+  [field-id]
+  (contains? (lib.metadata/field (qp.store/metadata-provider) field-id) :coercion-strategy))
```

Comment:
Is it possible to get coercion strategies from anywhere else than proper Fields?

Eg. a custom expression, or more likely in the metadata for a model?

Reply:
I think only fields can have coercion strategies.